pyproject.toml

```diff
@@ -0,0 +1,118 @@
+[tool.ruff]
+line-length = 100
+
+# Assume Python 3.11.
+target-version = "py311"
+
+[tool.ruff.lint]
+select = [
```

Comment:
I've enabled just about everything - may as well try to gradually improve all aspects of the codebase over time.